From a323a4bd1e72eea54b14291f409e8772bd9b7902 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Claudio=20Andr=C3=A9?= <claudioandre.br@gmail.com>
Date: Sat, 15 Dec 2018 23:35:51 -0200
Subject: [PATCH] Handle self confined system wide build.

---
 src/listconf.c | 12 ++++++++++++
 src/params.h   | 12 ++++++++++--
 2 files changed, 22 insertions(+), 2 deletions(-)

diff --git a/src/listconf.c b/src/listconf.c
index 319288ca7b..cc958100b9 100644
--- a/src/listconf.c
+++ b/src/listconf.c
@@ -155,6 +155,18 @@ static void listconf_list_build_info(void)
 	       SIMD_PARA_SHA256, SIMD_PARA_SHA512);
 #endif
 #if JOHN_SYSTEMWIDE
+#if defined(_BOXED)
+{
+	char *tmp_value;
+
+	if ((tmp_value = getenv("SNAP")) &&
+	     strstr(tmp_value, "/snap/john-the-ripper/"))
+		puts("Deploy: " "sandboxed as a Snap app");
+	else if ((tmp_value = getenv("FLATPAK_ID")) &&
+	          strstr(tmp_value, "com.openwall.John"))
+		puts("Deploy: " "sandboxed as a FlatPak app");
+}
+#endif
 	puts("System-wide exec: " JOHN_SYSTEMWIDE_EXEC);
 	puts("System-wide home: " JOHN_SYSTEMWIDE_HOME);
 	puts("Private home: " JOHN_PRIVATE_HOME);
diff --git a/src/params.h b/src/params.h
index 74726debd1..711b13dd21 100644
--- a/src/params.h
+++ b/src/params.h
@@ -104,10 +104,18 @@

 #if JOHN_SYSTEMWIDE
 #ifndef JOHN_SYSTEMWIDE_EXEC /* please refer to the notes above */
-#define JOHN_SYSTEMWIDE_EXEC		"/usr/libexec/john"
+#if defined(_SNAP)
+#define JOHN_SYSTEMWIDE_EXEC		"/snap/john-the-ripper/current/bin"
+#else
+#define JOHN_SYSTEMWIDE_EXEC		"/app/bin"
+#endif
 #endif
 #ifndef JOHN_SYSTEMWIDE_HOME
-#define JOHN_SYSTEMWIDE_HOME		"/usr/share/john"
+#if defined(_SNAP)
+#define JOHN_SYSTEMWIDE_HOME		"/snap/john-the-ripper/current/bin"
+#else
+#define JOHN_SYSTEMWIDE_HOME		"/app/bin"
+#endif
 #endif
 #define JOHN_PRIVATE_HOME		"~/.john"
 #endif
--
2.43.0
