From 0059f318d4fe4c29b2aa143aeca0af64dbc56ecc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Claudio=20Andr=C3=A9?= <dev@claudioandre.slmail.me>
Date: Sun, 1 Dec 2024 08:18:34 -0300
Subject: [PATCH] Remove peflags from the default target
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 7225b052f..38b36088f 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -194,7 +194,7 @@ default:
 	@$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) @CC_ASM_OBJS@"

-	@if [ "$(OS)" = "Windows_NT" ]; then \
+	@if [ "$(OS)" = "None-Windows_NT" ]; then \
 		$(MAKE) peflags; \
 	fi
 	@echo
--
2.43.0
